generators/app/index.js

```diff
@@ -174,13 +174,16 @@ module.exports = Generator.extend({
       // Allow distros to modify the composer.json.
       if (typeof options.drupalDistro.modifyComposer == 'function') {
         var done = this.async();
-        var self = this;
-        options.drupalDistro.modifyComposer(self, options, this.composer, isNewProject, done, function(err, result, done) {
+        options.drupalDistro.modifyComposer(options, this.composer, isNewProject, done, function(err, result, done) {
           if (!err && result) {
             this.composer = result;
           }
+          else {
+            this.log.warning("Could not retrieve Octane's composer.json: "  + err);
+            return done(err);
```

Comment:
By returning done() here, we make sure the following catch-all done() call is not reached. Extraneous done() calls are probably safe here, but they are a bad practice in general code as extra calls can lead to prematurely terminating an asynchronous process.